extension-style/src/main/java/com/mapbox/maps/extension/style/layers/properties/generated/Property.kt

```diff
@@ -533,6 +533,46 @@ enum class Anchor(override val value: String) : LayerProperty {
    */
   VIEWPORT("viewport"),
 }
+// NAME: The name of the projection to be used for rendering the map.
+/**
+ * The name of the projection to be used for rendering the map.
+ *
+ * @param value String value of this property
+ */
+enum class Name(override val value: String) : LayerProperty {
```

Comment:
```suggestion
enum class ProjectionName(override val value: String) : LayerProperty {
```

Reply:
this is generated code based on the style.. but guess it makes sense, I'll try to update generator code.

Reply:
sadly that will overcomplicate generator code... IMO we could leave this as is, we do not actually change property names for generated code in general from what I see. In general this enum will not be used standalone, it's used only in Style DSL + methods like `StyleInterface.getProjectionName` / `StyleInterface.setProjectionName`.

@mapbox/maps-android @mapbox/maps-ios - what are your thoughts here?

Reply:
`ProjectionName` does sound better than just `Name`. but if it will overcomplicate the generator code. we could leave it.

Reply:
@ank27 it's not only about naming here in the enum. We should re-organize generator code to understand this as well when generating methods `name()`in `Projection` file. Ideally we want something like:
```
Projection.name(name: ProjectionName)
```
but it's a bit tricky to achieve and we don't do this anywhere else.

@pengdev - as the most experienced person in generated code - what are your thoughts here?

Reply:
@kiryldz we can overwrite the style spec json in the style-parser.js, I think it makes sense to rename `name` to `projection_name`. We had similar tweaks to rename `GeojsonSource` to `GeoJsonSource` in the style-parser.js, you can use it as a reference.